models/db/index.go

```diff
@@ -47,6 +47,36 @@ func UpsertResourceIndex(ctx context.Context, tableName string, groupID int64) (
 	return err
 }
 
+// UpsertDecrResourceIndex the function will not return until it acquires the lock or receives an error.
+func UpsertDecrResourceIndex(ctx context.Context, tableName string, groupID, prevIdx int64) (err error) {
```

Comment:
For reviewers: Most of this code was copied from `UpsertResourceIndex` (which is a few lines above this function).